server/src/test/java/org/opensearch/rest/action/tiering/TieringRequestValidatorTests.java

```diff
@@ -0,0 +1,312 @@
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.tiering.TieringServiceValidator;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.opensearch.cluster.routing.allocation.DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING;
+import static org.opensearch.cluster.routing.allocation.DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING;
+import static org.opensearch.cluster.routing.allocation.DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING;
+
+public class TieringRequestValidatorTests extends OpenSearchTestCase {
+
+    private TieringServiceValidator validator = new TieringServiceValidator();
+
+    public void testValidateSearchNodes() {
+        ClusterState clusterStateWithSearchNodes = ClusterState.builder(ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
+            .nodes(createNodes(2, 0, 0))
+            .build();
+
+        // throws no errors
+        validator.validateSearchNodes(clusterStateWithSearchNodes, "test_index");
+    }
+
+    public void testValidateSearchNodesThrowError() {
+        ClusterState clusterStateWithNoSearchNodes = ClusterState.builder(ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
+            .nodes(createNodes(0, 1, 1))
+            .build();
+        // throws error
+        IllegalArgumentException e = expectThrows(
+            IllegalArgumentException.class,
+            () -> validator.validateSearchNodes(clusterStateWithNoSearchNodes, "test")
+        );
+    }
+
+    public void testValidRemoteStoreIndex() {
+        String indexUuid = UUID.randomUUID().toString();
+        String indexName = "test_index";
+
+        ClusterState clusterState1 = buildClusterState(
+            indexName,
+            indexUuid,
+            Settings.builder()
+                .put(IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING.getKey(), true)
+                .put(IndexMetadata.INDEX_REPLICATION_TYPE_SETTING.getKey(), ReplicationType.SEGMENT)
+                .build()
+        );
+
+        assertTrue(validator.validateRemoteStoreIndex(clusterState1, new Index(indexName, indexUuid)));
+    }
+
+    public void testNotRemoteStoreIndex() {
+        String indexUuid = UUID.randomUUID().toString();
+        String indexName = "test_index";
+        assertFalse(
+            validator.validateRemoteStoreIndex(buildClusterState(indexName, indexUuid, Settings.EMPTY), new Index(indexName, indexUuid))
+        );
+    }
+
+    public void testValidHotIndex() {
+        String indexUuid = UUID.randomUUID().toString();
+        String indexName = "test_index";
+        assertTrue(validator.validateHotIndex(buildClusterState(indexName, indexUuid, Settings.EMPTY), new Index(indexName, indexUuid)));
+    }
+
+    public void testNotHotIndex() {
```

Comment:
`testIndexWithOngoingTiering`

Reply:
renamed to `testIndexWithOngoingOrCompletedTiering`